### PR TITLE
Make TimeParamerization configurable

### LINKED ISSUE #330: Time between points is not strictly increasing
Time parameterization occasionally causes an error like this:

`Time between points 10 and 11 is not strictly increasing, it is 0.357595 and 0.357595 respectively`

For reference, #326 proposed a simple solution.

### PR DESCRIPTION
**Pull request by rhaschke**
Replaces #333, implementing https://github.com/ros-planning/moveit_task_constructor/pull/333#pullrequestreview-850576619. Fixes #330.
The TimeParameterization method becomes a property in corresponding stages and PlannerInterfaces.
This allows users to configure them easily.

### CODE REVIEW

File: core/src/merge.cpp
Comment (v4hn) [inline core/src/merge.cpp:169]: This can (and could before) break a lower specified dynamics limit in one of the merged trajectories. The merged trajectory might suddenly be "inexplicably" faster.

But the real solution to this is not adding yet another set of parameters, but implement a "proper" merge, that does not need to compute a whole new timing for the trajectory, but resamples the trajectories as needed.
So I guess it's best to leave this as is until then.

File: core/include/moveit/task_constructor/moveit_compat.h
Comment (v4hn) [inline core/include/moveit/task_constructor/moveit_compat.h:67]: I just saw you [backported the polymorphic base class change to melodic](https://github.com/ros-planning/moveit/commit/393815e4872de6349ae3784360dbe276a9abda7a). So this should be unnecessary.
But Jafar and you both did not notice the missing pointers in the original patch, so I just filed [this PR](https://github.com/ros-planning/moveit/pull/3078). It might be worth creating a patch release for melodic with this addition as well. If you should do that, these lines can go away
```suggestion
```

**Comment by rhaschke** [comment]
This commit finally breaks compatibility to older MoveIt releases, namely Melodic. How should we handle this? Drop Melodic? 
I'm fine with that.

**Review by v4hn — APPROVED** [review]
Mostly looks good. I'll leave it to you whether you want to deal with the missing pointers in upstream before merging.

I would be happier with this if ...
- the time parameterization were an actual plugin in MoveIt
- if the triple `<plugin, vel_scaling, acc_scaling>` would be somehow one property instead of three needed in multiple places
- if we could add additional string overloads to load these properties instead of having to pass shared_ptr.

File: core/src/stages/connect.cpp
Comment (rhaschke) [reply core/src/stages/connect.cpp]: Yeah, you are right.

File: core/src/merge.cpp
Comment (rhaschke) [reply core/src/merge.cpp:169]: Please file an issue as a reminder...

File: core/include/moveit/task_constructor/moveit_compat.h
Comment (rhaschke) [reply core/include/moveit/task_constructor/moveit_compat.h:67]: Until we have a new release of MoveIt, this addition is required. Do you propose a patch release of MoveIt before? 
I could do that over the weekend indeed...

Comment (v4hn) [reply core/include/moveit/task_constructor/moveit_compat.h:67]: I guess you can remove this now. :)

**Review by JafarAbdi — APPROVED** [review]

**Pull request merged**
